metrics-proxy/src/main/java/ai/vespa/metricsproxy/metric/model/json/JacksonUtil.java

```diff
@@ -19,12 +20,18 @@
  */
 public class JacksonUtil {
 
-    private static final DecimalFormat decimalFormat = createDecimalFormat();
-    private static DecimalFormat createDecimalFormat() {
+    private static final ThreadLocal<DecimalFormat> withinLongRangeFormat = ThreadLocal.withInitial(JacksonUtil::createWithinLongRangeFormat);
+    private static final ThreadLocal<DecimalFormat> outsideLongRangeFormat = ThreadLocal.withInitial(JacksonUtil::createOutsideLongRangeFormat);
```

Comment:
Just curious, why do these have to be ThreadLocal?

Reply:
It was me wanted to be on the safe side due to the below comment in DecimalFormat.java.
From DecimalFormat.java
Synchronization
Decimal formats are generally not synchronized. It is recommended to create separate format instances for each thread. If multiple threads access a format concurrently, it must be synchronized externally.